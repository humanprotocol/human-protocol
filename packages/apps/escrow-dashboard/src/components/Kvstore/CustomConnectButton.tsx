import Button from '@mui/material/Button';
import Avatar from '@mui/material/Avatar';
import Box from '@mui/material/Box';
import ArrowDropDownIcon from '@mui/icons-material/ArrowDropDown';
import { ConnectButton } from '@rainbow-me/rainbowkit';
export const CustomConnectButton = () => {
  return (
    <ConnectButton.Custom>
      {({ account, chain, openAccountModal, openConnectModal, mounted }) => {
        // Note: If your app doesn't use authentication, you
        // can remove all 'authenticationStatus' checks
        const ready = mounted;
        const connected = ready && account && chain;
        return (
          <div
            {...(!ready && {
              'aria-hidden': true,
              style: {
                opacity: 0,
                pointerEvents: 'none',
                userSelect: 'none',
              },
            })}
          >
            {(() => {
              if (!connected) {
                return (
                  <Button
                    sx={{ marginLeft: { xs: 0.7, sm: 0.7, md: 0 } }}
                    onClick={openConnectModal}
                    variant="outlined"
                  >
                    Connect Wallet
                  </Button>
                );
              }
              if (chain.unsupported) {
                return (
                  <Button onClick={openConnectModal} variant="outlined">
                    Wrong Network
                  </Button>
                );
              }
              return (
                <Box sx={{ display: 'flex', gap: 12 }}>
                  <Button
                    endIcon={<ArrowDropDownIcon />}
                    onClick={openAccountModal}
                    variant="contained"
                    sx={{
                      '&:hover': {
                        color: 'gray',
                        backgroundColor: '#f6f7fd',
                      },
                      borderRadius: 12,
                      paddingLeft: 1,
                      paddingTop: 0.4,
                      paddingBottom: 0.4,
                      backgroundColor: '#f6f7fe',
                      color: '#320a8d',
                    }}
                    startIcon={
                      <Avatar
                        src={
                          'https://cdn.stamp.fyi/avatar/eth:' +
                          account.displayName
                        }
                      />
                    }
                  >
                    {' '}
                    {account.displayName}
                  </Button>
                </Box>
              );
            })()}
          </div>
        );
      }}
    </ConnectButton.Custom>
  );
};
