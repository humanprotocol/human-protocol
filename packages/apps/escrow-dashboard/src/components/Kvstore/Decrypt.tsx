import KVStore from '@human-protocol/core/abis/KVStore.json';
import { ChainId, NETWORKS } from '@human-protocol/sdk';
import { Close as CloseIcon } from '@mui/icons-material';
import {
  Box,
  Button,
  Divider,
  Grid,
  Paper,
  Snackbar,
  TextField,
  Typography,
} from '@mui/material';
import * as openpgp from 'openpgp';
import { useState, ChangeEvent, FC } from 'react';
import { useContractRead, useAccount, useNetwork } from 'wagmi';

import { Alert } from '../Alert';
import { showIPFS } from 'src/services';

export const Decrypt: FC = () => {
  const { chain } = useNetwork();
  const [key, setKey] = useState('');
  const [passphrase, setPassphrase] = useState('');
  const [value, setValue] = useState('');
  const [error, setError] = useState('');
  const [privkey, setPrivkey] = useState('');
  const [decrypted, setDecrypted] = useState('');
  const [copy, setCopy] = useState<boolean>(false);
  const [loading, setLoading] = useState<boolean>(false);
  const [filename, setFilename] = useState<string>('');

  const { address } = useAccount();
  const { refetch } = useContractRead({
    address: NETWORKS[chain?.id as ChainId]?.kvstoreAddress as `0x${string}`,
    abi: KVStore,
    functionName: 'get',
    args: [address, key],
  });

  async function getValue() {
    setLoading(true);
    try {
      const { data } = await refetch();
      setValue(await showIPFS(data!.toString()));
      setLoading(false);
    } catch (e) {
      if (e instanceof Error) {
        setLoading(false);
        setError(e.message);
      }
    }
  }

  const handleFile = async (e: ProgressEvent<FileReader>) => {
    setLoading(true);
    setPrivkey('');

    try {
      const content = e.target!.result as string;
      const a = await openpgp.readKey({ armoredKey: content });
      if (!a.isPrivate()) {
        setError('Error, file is not private key');
        setLoading(false);
        setFilename('');
        return;
      }
      setPrivkey(content);
      setLoading(false);
    } catch (e) {
      if (e instanceof Error) {
        setError(e.message);
        setLoading(false);
        setFilename('');
      }
    }
  };

  const handleChange = (ev: ChangeEvent<HTMLInputElement>) => {
    const reader = new FileReader();
    reader.addEventListener('load', handleFile);
    reader.readAsText(ev.target.files![0]);
    setFilename(ev.target.files![0].name);
  };

  async function decryptValue() {
    setLoading(true);
    try {
      const privateKey = await openpgp.decryptKey({
        privateKey: await openpgp.readPrivateKey({ armoredKey: privkey }),
        passphrase,
      });

      const message = await openpgp.readMessage({
        armoredMessage: value,
      });

      const { data: decryptedVal } = await openpgp.decrypt({
        message,
        decryptionKeys: privateKey,
      });

      setDecrypted(decryptedVal as string);
      setLoading(false);
    } catch (e) {
      if (e instanceof Error) {
        setDecrypted('');
        setError(e.message);
        setLoading(false);
      }
    }
  }
  return (
    <Grid
      item
      xs={12}
      sm={12}
      md={12}
      container
      direction="column"
      justifyContent="center"
      alignItems="center"
    >
      <Snackbar
        anchorOrigin={{ vertical: 'top', horizontal: 'center' }}
        open={copy}
        autoHideDuration={3000}
        onClose={() => setCopy(false)}
      >
        <Alert
          onClose={() => setCopy(false)}
          severity="info"
          sx={{ width: '100%' }}
        >
          copied
        </Alert>
      </Snackbar>

      <Snackbar
        anchorOrigin={{ vertical: 'top', horizontal: 'center' }}
        open={error.length > 0}
        autoHideDuration={3000}
        onClose={() => setError('')}
      >
        <Alert
          onClose={() => setError('')}
          severity="error"
          sx={{ width: '100%' }}
        >
          {error}
        </Alert>
      </Snackbar>
      <Box sx={{ width: { xs: 1, md: '50%', lg: '50%', xl: '40%' } }}>
        <Paper>
          <Box sx={{ borderBottom: '1px solid #CBCFE6' }}>
            <Grid
              container
              direction="row"
              justifyContent={'space-between'}
              flexDirection={{ xs: 'column-reverse', md: 'row' }}
            >
              <Grid item>
                <Typography fontWeight={`500`} padding={`10px`}>
                  Decrypt
                </Typography>
              </Grid>
            </Grid>
          </Box>
          <Grid
            container
            direction="column"
            sx={{
              padding: {
                xs: '0px 80px 0px 80px',
                md: '0px 80px 0px 80px',
                lg: '0px 80px 0px 80px',
                xl: '0px 80px 0px 80px',
              },
            }}
          >
            <Grid
              item
              container
              direction="column"
              sx={{
                marginBottom: { lg: 10 },
              }}
            >
              <Box sx={{ mt: { md: 10, lg: 10, xl: 10 }, mb: 2 }}>
                <Typography>
                  Decrypt using your public key and private key
                </Typography>
              </Box>

              <TextField
                sx={{ my: 1 }}
                id="outlined-basic"
                label="Key"
                variant="outlined"
                value={key}
                onChange={(e) => setKey(e.target.value)}
                disabled={loading}
              />
              <Box>
                <Button
                  disabled={loading}
                  onClick={getValue}
                  sx={{ my: 1, mb: !value ? 5 : 2 }}
                  variant="outlined"
                >
                  Get
                </Button>
              </Box>
              {value && (
                <>
                  <Box>
                    <Typography fontWeight={500}>Value</Typography>
                  </Box>
                  <Box sx={{ width: { xs: 1 } }}>
                    <Box
                      className="pubkey"
                      sx={{
                        backgroundColor: '#f6f7fe',
                        maxHeight: 200,
                        marginTop: 2,
                        overflowY: 'scroll',
                        overflowWrap: 'break-word',
                        padding: 4,
                        borderRadius: 3,
                      }}
                    >
                      {value}
                    </Box>
                  </Box>
                  <Box sx={{ marginBottom: { xs: 2, sm: 2, md: 2, lg: 0 } }}>
                    <Button
                      onClick={() => {
                        setCopy(true);
                        navigator.clipboard.writeText(value);
                      }}
                      size="small"
                    >
                      Copy
                    </Button>
                  </Box>
                  <Divider sx={{ my: 2 }} light />
                  <Box>
                    <Typography>Decrypt </Typography>
                  </Box>
                  <Box
                    display={`flex`}
                    flexDirection={'row'}
                    alignContent={`center`}
                  >
                    {privkey.length === 0 && (
                      <Button
                        sx={{ my: 2 }}
                        disabled={loading}
                        variant="outlined"
                        size="small"
                        component="label"
                      >
                        Upload Private Key
                        <input
                          type="file"
                          aria-label="add files"
                          hidden
                          onChange={handleChange}
                        />
                      </Button>
                    )}
                    {privkey.length > 0 && (
                      <>
                        <Typography sx={{ my: 2 }} variant={`body1`}>
                          {filename}
                        </Typography>
                        <CloseIcon
                          onClick={() => setPrivkey('')}
                          sx={{ cursor: 'pointer', my: 2.3 }}
                        />
                      </>
                    )}
                  </Box>
                  <TextField
                    sx={{ my: 1 }}
                    id="outlined-basic"
                    label="Passphrase"
                    variant="outlined"
                    value={passphrase}
                    type="password"
                    disabled={loading}
                    onChange={(e) => setPassphrase(e.target.value)}
                  />
                  <Box
                    display={`flex`}
                    width={`100%`}
                    justifyContent={`flex-end`}
                  >
                    <Button
                      disabled={loading}
                      sx={{ my: 1, mb: !decrypted ? 5 : 2 }}
                      variant="contained"
                      onClick={decryptValue}
                    >
                      Decrypt
                    </Button>
                  </Box>
                  {decrypted && (
                    <>
                      <Box>
                        <Typography fontWeight={`500`}>
                          Decrypted Value{' '}
                        </Typography>
                      </Box>
                      <Box
                        className="pubkey"
                        sx={{
                          backgroundColor: '#f6f7fe',
                          maxHeight: 200,
                          marginTop: 2,
                          overflowY: 'scroll',
                          overflowWrap: 'break-word',
                          padding: 4,
                          borderRadius: 3,
                        }}
                      >
                        {decrypted}
                      </Box>
                      <Box
                        sx={{ marginBottom: { xs: 2, sm: 2, md: 2, lg: 0 } }}
                      >
                        <Button
                          onClick={() => {
                            setCopy(true);
                            navigator.clipboard.writeText(value);
                          }}
                          size="small"
                        >
                          Copy
                        </Button>
                      </Box>
                    </>
                  )}
                </>
              )}
            </Grid>
          </Grid>
        </Paper>
      </Box>
    </Grid>
  );
};
