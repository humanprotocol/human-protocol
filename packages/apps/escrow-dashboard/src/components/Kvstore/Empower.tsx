import { Grid, Paper, Typography, Box, Button } from '@mui/material';
import DoneIcon from '@mui/icons-material/Done';
import React, { Dispatch } from 'react';

export const Empower = ({
  refetch,
  setPublicKey,
}: {
  refetch: any;
  setPublicKey: Dispatch<string>;
}): React.ReactElement => {
  async function finish() {
    try {
      const { data } = await refetch();
      setPublicKey(data);
    } catch (e) {
      if (e instanceof Error) {
      }
    }
  }
  return (
    <Grid container>
      <Grid item container direction="row" justifyContent="center">
        <Grid
          item
          xs={12}
          sm={12}
          md={12}
          container
          direction="column"
          justifyContent="center"
          alignItems="center"
        >
          <Box>
            <Paper>
              <Grid container direction="column">
                <Grid
                  item
                  container
                  direction="column"
                  alignItems="center"
                  sx={{
                    marginTop: { xs: 1, sm: 1, md: 10, lg: 10 },
                    paddingLeft: 10,
                    paddingRight: 10,
                    marginBottom: { md: 10, lg: 10 },
                  }}
                >
                  <DoneIcon />
                  <Typography
                    sx={{ marginTop: 3 }}
                    variant="h6"
                    color="primary"
                  >
                    Success!
                  </Typography>
                  <Typography
                    sx={{ marginTop: 1 }}
                    variant="body2"
                    color="primary"
                  >
                    Your public key has been stored on the blockchain now, click
                    on continue to encrypt or decrypt
                  </Typography>
                  <Button
                    onClick={finish}
                    variant="contained"
                    sx={{ marginTop: 2 }}
                  >
                    Continue
                  </Button>
                </Grid>
              </Grid>
            </Paper>
          </Box>
        </Grid>
      </Grid>
    </Grid>
  );
};
