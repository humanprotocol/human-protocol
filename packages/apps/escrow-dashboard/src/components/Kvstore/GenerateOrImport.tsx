import { Grid, Paper, Typography, Button } from '@mui/material';
import { Dispatch, FC } from 'react';

import keySvg from 'src/assets/key.svg';

export type GenerateOrImportProps = {
  setStep: Dispatch<number>;
  setPage: Dispatch<number>;
};

export const GenerateOrImport: FC<GenerateOrImportProps> = ({
  setStep,
  setPage,
}) => (
  <Paper>
    <Grid container justifyContent="center" direction="column">
      <Grid
        item
        container
        direction="column"
        alignItems="center"
        sx={{
          marginTop: { xs: 1, sm: 1, md: 10, lg: 10 },
          marginBottom: { lg: 10 },
        }}
      >
        {' '}
        <img width="100" src={keySvg} alt="lbank" />{' '}
        <Typography sx={{ marginTop: 3 }} variant="body2" color="primary">
          If you already have your public key import it, if not generate it
        </Typography>
      </Grid>
      <Grid
        item
        container
        direction="row"
        justifyContent="center"
        alignItems="flex-end"
        sx={{
          marginTop: { xs: 1, sm: 1, md: 10, lg: 10 },

          marginBottom: { xs: 1, sm: 1, md: 10, lg: 7 },
        }}
      >
        <Button
          onClick={() => {
            setStep(1);
            setPage(1);
          }}
          variant="contained"
          sx={{ marginRight: 2 }}
        >
          Generate
        </Button>
        <Button
          onClick={() => {
            setStep(1);
            setPage(2);
          }}
          variant="outlined"
        >
          Import
        </Button>
      </Grid>
    </Grid>
  </Paper>
);
