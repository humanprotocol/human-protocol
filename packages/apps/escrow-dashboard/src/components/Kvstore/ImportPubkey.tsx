import {
  Key as KeyIcon,
  UploadFile as UploadFileIcon,
} from '@mui/icons-material';
import {
  Box,
  Button,
  Grid,
  List,
  ListItem,
  ListItemButton,
  ListItemIcon,
  ListItemText,
  ListSubheader,
  Paper,
  Snackbar,
  Typography,
  TextField,
} from '@mui/material';
import { readKey } from 'openpgp/lightweight';
import { Dispatch, useState, ChangeEvent, FC } from 'react';

import { Alert } from '../Alert';
import { Key } from './types';
import { showGithubPGP } from 'src/services';

const handleFile = async (
  e: ProgressEvent<FileReader>,
  setError: Dispatch<string>,
  setKey: Dispatch<Key>,
  setPage: Dispatch<number>,
  setStep: Dispatch<number>
) => {
  try {
    const pubkey = e.target!.result as string;
    const a = await readKey({ armoredKey: pubkey });
    if (a.isPrivate()) {
      setError('Error, file is not public key');
      return;
    }
    setKey({ publicKey: pubkey, privateKey: '' });
    setPage(2.5);
    setStep(1);
  } catch (e) {
    if (e instanceof Error) setError("Error, file doesn't contain public key");
  }
};

const handleChange = (
  ev: ChangeEvent<HTMLInputElement>,
  setError: Dispatch<string>,
  setKey: Dispatch<Key>,
  setPage: Dispatch<number>,
  setStep: Dispatch<number>
) => {
  const reader = new FileReader();
  reader.addEventListener('load', (e) =>
    handleFile(e, setError, setKey, setPage, setStep)
  );
  reader.readAsText(ev.target.files![0]);
};

function pickPublicKey(
  gpg: string,
  setKey: Dispatch<Key>,
  setPage: Dispatch<number>,
  setStep: Dispatch<number>
) {
  setKey({ publicKey: gpg, privateKey: '' });
  setPage(2.5);
  setStep(1);
}

async function getGithubGPGFile(
  username: string,
  setError: Dispatch<string>,
  setPublicKeys: Dispatch<
    Array<{
      name: string;
      raw_key: string;
      key_id: string;
    }>
  >
) {
  try {
    const gpg = await showGithubPGP(username);
    const parsedGPG = JSON.parse(gpg);
    if (parsedGPG.length === 0) {
      setError("Error, user doesn't have any public key");
      return;
    }

    setPublicKeys(parsedGPG);
  } catch (e) {
    if (e instanceof Error) setError("Error, can't get github GPG file");
  }
}

export type ImportPubkeyProps = {
  setStep: Dispatch<number>;
  setPage: Dispatch<number>;
  setKey: Dispatch<Key>;
  pubkeyExist: boolean;
  refetch: any;
  setPublicKey: Dispatch<string>;
};

export const ImportPubkey: FC<ImportPubkeyProps> = ({
  setStep,
  setPage,
  setKey,
  pubkeyExist,
  refetch,
  setPublicKey,
}) => {
  const [choose, setChoose] = useState<string>('');
  const [error, setError] = useState<string>('');
  const [username, setUsername] = useState<string>('');
  const [publicKeys, setPublicKeys] = useState<
    Array<{ raw_key: string; key_id: string; name: string }>
  >([]);

  async function goBack() {
    try {
      if (pubkeyExist) {
        const { data } = await refetch();
        setPublicKey(data);
      } else {
        if (choose.trim().length !== 0) {
          setChoose('');
        } else {
          setStep(0);
          setPage(0);
        }
      }
    } catch (e) {
      if (e instanceof Error) {
        setError(e.message);
      }
    }
  }

  return (
    <Paper>
      <Snackbar
        anchorOrigin={{ vertical: 'top', horizontal: 'center' }}
        open={error.length > 0}
        autoHideDuration={6000}
        onClose={() => setError('')}
      >
        <Alert
          onClose={() => setError('')}
          severity="error"
          sx={{ width: '100%' }}
        >
          {error}
        </Alert>
      </Snackbar>
      <Grid container direction="column">
        <Grid
          item
          container
          direction="column"
          alignItems="flex-start"
          justifyContent="center"
          sx={{ paddingLeft: 5, paddingTop: 5 }}
        >
          {choose === 'file' && (
            <>
              <Typography variant="body2" color="primary">
                Add your public key file
              </Typography>
              <Grid
                container
                sx={{
                  my: { lg: 1, xl: 1 },
                  paddingRight: { xs: 1, sm: 1, md: 5, lg: 5 },
                }}
              >
                <Paper
                  sx={{
                    backgroundColor: '#f6f7fe',
                    width: { xs: 220, md: 600, lg: 600, xl: 600 },
                    padding: 2,
                    marginTop: 2,
                  }}
                >
                  <Grid container justifyContent="center">
                    <Button
                      variant="outlined"
                      component="label"
                      startIcon={<UploadFileIcon />}
                    >
                      Select or drop file
                      <input
                        type="file"
                        aria-label="add files"
                        hidden
                        onChange={(e) =>
                          handleChange(e, setError, setKey, setPage, setStep)
                        }
                      />
                    </Button>
                  </Grid>
                </Paper>
              </Grid>
            </>
          )}
          {choose === 'github' && (
            <>
              <Typography variant="body2" color="primary">
                Add your public key file from github
              </Typography>
              <Grid
                container
                sx={{
                  my: { lg: 1, xl: 1 },
                  paddingRight: { xs: 1, sm: 1, md: 5, lg: 5 },
                }}
              >
                <Paper
                  sx={{
                    backgroundColor: '#f6f7fe',
                    width: { xs: 220, md: 600, lg: 600, xl: 600 },
                    padding: 2,
                    marginTop: 2,
                  }}
                >
                  <Grid container justifyContent="center">
                    {publicKeys.length > 0 && (
                      <Box
                        sx={{
                          width: '100%',
                          maxWidth: 360,
                          bgcolor: 'background.paper',
                        }}
                      >
                        <List
                          subheader={
                            <ListSubheader
                              component="div"
                              id="nested-list-subheader"
                            >
                              Pick your public key
                            </ListSubheader>
                          }
                        >
                          {publicKeys.map((a) => (
                            <ListItem key={a.raw_key} disablePadding>
                              <ListItemButton
                                onClick={() =>
                                  pickPublicKey(
                                    a.raw_key,
                                    setKey,
                                    setPage,
                                    setStep
                                  )
                                }
                              >
                                <ListItemIcon>
                                  <KeyIcon />
                                </ListItemIcon>
                                <Grid justifyContent={`center`}>
                                  <ListItemText primary={a?.name} />
                                  <ListItemText
                                    primary={`Key ID : ${a.key_id}`}
                                  />
                                </Grid>
                              </ListItemButton>
                            </ListItem>
                          ))}
                        </List>
                      </Box>
                    )}

                    {publicKeys.length === 0 && (
                      <TextField
                        sx={{ width: { lg: 200, xl: 200 } }}
                        id="outlined-basic"
                        label="Github username"
                        variant="outlined"
                        value={username}
                        onChange={(e) => setUsername(e.target.value)}
                      />
                    )}
                  </Grid>
                  <Grid container justifyContent="center">
                    {publicKeys.length === 0 ? (
                      <Button
                        variant="outlined"
                        onClick={() =>
                          getGithubGPGFile(username, setError, setPublicKeys)
                        }
                        sx={{ mx: 2, mt: 2 }}
                      >
                        Get public key
                      </Button>
                    ) : (
                      <Button
                        variant="outlined"
                        onClick={() => {
                          setChoose('github');
                          setPublicKeys([]);
                        }}
                        sx={{ mx: 2, mt: 2 }}
                      >
                        Try another username
                      </Button>
                    )}
                  </Grid>
                </Paper>
              </Grid>
            </>
          )}
          {choose.length === 0 && (
            <>
              {' '}
              <Typography variant="body2" color="primary">
                Import your public key
              </Typography>
              <Grid
                container
                sx={{
                  width: { md: 650, lg: 650, xl: 650 },
                  my: { lg: 1, xl: 1 },
                  paddingRight: { xs: 1, sm: 1, md: 5, lg: 5 },
                }}
              >
                <Grid container justifyContent="center">
                  <Button
                    onClick={() => setChoose('file')}
                    variant="contained"
                    sx={{ mx: 2, mt: 2 }}
                  >
                    Using file
                  </Button>
                </Grid>
                <Grid container justifyContent="center">
                  <Button
                    onClick={() => setChoose('github')}
                    variant="contained"
                    sx={{ mx: 2, mt: 2 }}
                  >
                    Using github
                  </Button>
                </Grid>
              </Grid>
            </>
          )}
        </Grid>

        <Grid
          item
          container
          direction="row"
          justifyContent="flex-end"
          alignItems="flex-end"
          sx={{
            marginTop: { xs: 1, sm: 1, md: 10, lg: 10 },
            paddingRight: { xs: 1, sm: 1, md: 5, lg: 5 },
            marginBottom: { xs: 1, sm: 1, md: 7, lg: 7 },
          }}
        >
          <Button variant="outlined" sx={{ mr: 2 }} onClick={goBack}>
            Back
          </Button>
        </Grid>
      </Grid>
    </Paper>
  );
};
