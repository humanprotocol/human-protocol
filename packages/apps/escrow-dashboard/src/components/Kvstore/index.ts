export { KvstoreView } from './KvstoreView';
export type Key = {
  publicKey: string;
  privateKey: string;
};
