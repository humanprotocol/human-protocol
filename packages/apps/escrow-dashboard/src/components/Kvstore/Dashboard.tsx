import { Grid, Box, Tabs, Tab } from '@mui/material';
import React, { useState, useEffect, Dispatch } from 'react';
import ViewTitle from '../ViewTitle';

import { StoredPubkey } from './StoredPubkey';
import { Decrypt } from './Decrypt';
import { Encrypt } from './Encrypt';
import { showIPFS } from '../../services/index';

export const Dashboard = ({
  publicKey,
  refetch,
  setPublicKey,
  setStep,
  setPage,
}: {
  publicKey: string;
  refetch: any;
  setPublicKey: Dispatch<string>;
  setStep: Dispatch<number>;
  setPage: Dispatch<number>;
}): React.ReactElement => {
  const [value, setValue] = React.useState(0);
  const [pubkey, setPubkey] = useState<string>('');
  useEffect(() => {
    if (publicKey.trim().length !== 0) {
      showIPFS(publicKey).then((a) => setPubkey(a));
    }
  }, [publicKey]);
  const handleChange = (event: React.SyntheticEvent, newValue: number) => {
    setValue(newValue);
  };
  return (
    <Grid container>
      <Grid
        item
        xs={12}
        sm={12}
        md={12}
        direction="row"
        justifyContent="center"
      >
        <Box
          marginBottom="1em"
          display="flex"
          alignItems="center"
          flexWrap="wrap"
        >
          <ViewTitle title="ETH KV Store" iconUrl="/images/small_key.svg" />
        </Box>
        <Box sx={{ borderBottom: 1, borderColor: 'divider', marginBottom: 2 }}>
          {pubkey.trim().length > 0 && (
            <Tabs value={value} onChange={handleChange}>
              <Tab label="Public Key" />
              <Tab label="Encrypt" />
              <Tab label="Decrypt" />
            </Tabs>
          )}
        </Box>
        {value === 0 && (
          <StoredPubkey
            refetch={refetch}
            publicKey={pubkey}
            setStep={setStep}
            setPage={setPage}
            setPublicKey={setPublicKey}
          />
        )}
        {value === 1 && <Encrypt publicKey={pubkey} />}
        {value === 2 && <Decrypt />}
      </Grid>
    </Grid>
  );
};
