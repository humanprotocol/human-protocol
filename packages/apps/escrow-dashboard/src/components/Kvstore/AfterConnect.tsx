import {
  Box,
  Grid,
  Paper,
  Step,
  Stepper,
  StepLabel,
  Typography,
} from '@mui/material';
import { useState, Dispatch, FC } from 'react';

import { STEPS } from './constants';
import { Empower } from './Empower';
import { GenerateOrImport } from './GenerateOrImport';
import { GeneratePubkey } from './GeneratePubkey';
import { ImportPubkey } from './ImportPubkey';
import { Success } from './Success';
import { Key } from './types';

type StepPage = {
  step: number;
  setStep: Dispatch<number>;
  page: number;
  setPage: Dispatch<number>;
};

export type AfterConnectProps = {
  pubkeyExist: boolean;
  refetch: any;
  setPublicKey: Dispatch<string>;
} & StepPage;

export const AfterConnect: FC<AfterConnectProps> = ({
  step,
  setStep,
  page,
  setPage,
  pubkeyExist,
  refetch,
  setPublicKey,
}) => {
  const [key, setKey] = useState<Key>({ publicKey: '', privateKey: '' });

  return (
    <Grid container>
      <Grid
        item
        xs={12}
        sm={12}
        md={12}
        container
        direction="row"
        justifyContent="center"
      >
        <Grid
          item
          xs={12}
          sm={12}
          md={12}
          container
          direction="column"
          justifyContent="center"
          alignItems="center"
        >
          <Box>
            <Typography variant="h4" sx={{ marginBottom: 2 }} color="primary">
              ETH KV Store
            </Typography>
            <Paper sx={{ padding: { md: 2 }, marginBottom: 2 }}>
              {' '}
              <Box sx={{ width: '100%' }}>
                <Stepper activeStep={step}>
                  {STEPS.map((step) => (
                    <Step key={step}>
                      <StepLabel>{step}</StepLabel>
                    </Step>
                  ))}
                </Stepper>
              </Box>
            </Paper>

            {page === 0 && (
              <GenerateOrImport setStep={setStep} setPage={setPage} />
            )}
            {page === 1 && (
              <GeneratePubkey
                refetch={refetch}
                setPublicKey={setPublicKey}
                pubkeyExist={pubkeyExist}
                setKey={setKey}
                setStep={setStep}
                setPage={setPage}
              />
            )}
            {page === 1.5 && (
              <Success
                what="generated"
                keys={key}
                setStep={setStep}
                setPage={setPage}
              />
            )}
            {page === 2 && (
              <ImportPubkey
                refetch={refetch}
                setPublicKey={setPublicKey}
                pubkeyExist={pubkeyExist}
                setKey={setKey}
                setStep={setStep}
                setPage={setPage}
              />
            )}
            {page === 2.5 && (
              <Success
                what="imported"
                keys={key}
                setStep={setStep}
                setPage={setPage}
              />
            )}
            {page === 3 && (
              <Empower refetch={refetch} setPublicKey={setPublicKey} />
            )}
          </Box>
        </Grid>
      </Grid>
    </Grid>
  );
};
