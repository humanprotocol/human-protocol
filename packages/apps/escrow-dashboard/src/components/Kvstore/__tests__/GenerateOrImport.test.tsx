import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { act } from 'react-dom/test-utils';
import { GenerateOrImport } from 'src/components/Kvstore/GenerateOrImport';

import { RainbowKitProvider } from '@rainbow-me/rainbowkit';
import {
  Providers,
  setupClient,
  getSigners,
  testChains,
} from '../../../../tests/utils';
import { MockConnector } from '@wagmi/core/connectors/mock';

describe('when rendered GenerateOrImport component', () => {
  it('should render `text` prop', async () => {
    const client = setupClient({
      connectors: [
        new MockConnector({
          options: {
            signer: getSigners()[0]!,
            // Turn on `failConnect` flag to simulate connect failure
          },
        }),
      ],
    });
    await act(async () => {
      render(<GenerateOrImport />, {
        wrapper: ({ children }: { children: React.ReactNode }) => (
          <Providers client={client}>
            <RainbowKitProvider chains={testChains} modalSize="compact">
              {children}
            </RainbowKitProvider>
          </Providers>
        ),
      });
    });
    expect(screen.getByText(/Import/)).toBeInTheDocument();
  });
});

it('GenerateOrImport component renders correctly, corresponds to the snapshot', () => {
  const client = setupClient({
    connectors: [
      new MockConnector({
        options: {
          signer: getSigners()[0]!,
          // Turn on `failConnect` flag to simulate connect failure
        },
      }),
    ],
  });
  const tree = renderer
    .create(
      <Providers client={client}>
        <RainbowKitProvider chains={testChains} modalSize="compact">
          <GenerateOrImport />
        </RainbowKitProvider>
      </Providers>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
