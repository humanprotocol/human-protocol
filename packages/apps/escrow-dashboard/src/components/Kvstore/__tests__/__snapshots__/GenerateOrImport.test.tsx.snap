// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GenerateOrImport component renders correctly, corresponds to the snapshot 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1ps6pg7-MuiPaper-root"
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-6gh8l0-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column css-otsqzn-MuiGrid-root"
    >
       
      <img
        alt="lbank"
        src="/src/assets/key.svg"
        width="100"
      />
       
      <p
        className="MuiTypography-root MuiTypography-body2 css-1sn46sq-MuiTypography-root"
      >
        If you already have your public key import it, if not generate it
      </p>
    </div>
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-item css-1qch9dp-MuiGrid-root"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1s5vo3j-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Generate
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Import
      </button>
    </div>
  </div>
</div>
`;
